lib/std/fmt.zig

```diff
@@ -901,11 +901,18 @@ fn formatSizeImpl(comptime base: comptime_int) type {
             writer: anytype,
         ) !void {
             _ = fmt;
+            // The worst case for the calculated float value, plus three bytes for the suffix.
+            // TODO: Fix: if options.precision is high enough, this buffer will not be sufficient.
```

Comment:
Would be good to create an issue for this with a reproduction, especially if it affects all callsites of `formatFloat` in `std.fmt`.

Reply:
I'll do that right now.

Reply:
I also just realized that values under `base` do not print with the correct amount of decimal places. i can quickly fix that

Reply:
actually, do you think this is the right thing to do? should whole numbers keep the right number of decimal places, or should all whole numbers truncate the decimal points like 0 used to?

Reply:
It does seem like not printing the decimal places was intentional for sizes printed as bytes, judging from this test code:

https://github.com/ziglang/zig/blob/7d54c62c8a55240bbe144ab03c78573a344598ce/lib/std/fmt.zig#L2128-L2129

Maybe instead of the changes in this PR, this behavior should just be documented? Something like:

```
/// `precision` is ignored when `value` is less than 1 kB.
```

for `fmtIntSizeDec`, and

```
/// `precision` is ignored when `value` is less than 1 KiB.
```

for `fmtIntSizeBin`

Reply:
i just realized what i did broke some tests. that was probably intended behavior. sigh.

Reply:
You're right, I guess the whole premise of this pull request was kind of useless :sweat_smile:, I suppose I can just undo everything and submit the pull request with only those comments 

oh well, misunderstandings happen

Reply:
I undid my changes and just added those doc comments. I'm sorry for wasting your time because of a misunderstanding, lol